### PR TITLE
Add Spotless for code & commit style checks

### PR DESCRIPTION
**Pull request by goonerDroid**
- Add [Spotless](https://github.com/diffplug/spotless) v.6.25.0 with [kt-fmt](https://github.com/facebook/ktfmt) 
- Update `.gitignore`

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,4 +1,27 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!--
```
Comment (darshanpania) [inline app/src/main/AndroidManifest.xml:2]: @goonerDroid Why is MIT licence needed in the manifest file?

File: app/src/main/java/com/darshan/notificity/AppDatabase.kt [hunk 2]
```diff
@@ -1,3 +1,27 @@
+/*
```
Comment (darshanpania) [inline app/src/main/java/com/darshan/notificity/AppDatabase.kt:1]: @goonerDroid Why is the MIT License being added to all the files?

File: .scripts/commit-rule [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (darshanpania) [inline .scripts/commit-rule:1]: @goonerDroid Can you remove this for now? I will add this rule via GitHub settings

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (goonerDroid) [reply app/src/main/AndroidManifest.xml:2]: @darshanpania I have Added a license script for all files, and this license template will be added during spotlessTask. Do you want me to exclude the license from the manifest(.xml) or remove it completely from .xml files?

File: app/src/main/java/com/darshan/notificity/AppDatabase.kt (hunk 2 above)
Comment (goonerDroid) [reply app/src/main/java/com/darshan/notificity/AppDatabase.kt:1]: @darshanpania All new files which will be created/added will automatically have the open-source license by default..hence added the script

File: .scripts/commit-rule (hunk 3 above)
Comment (goonerDroid) [reply .scripts/commit-rule:1]: @goonerDroid Ack. Will remove `commit-rule`(lock commits on the main branch)

File: app/src/main/java/com/darshan/notificity/AppDatabase.kt (hunk 2 above)
Comment (darshanpania) [reply app/src/main/java/com/darshan/notificity/AppDatabase.kt:1]: @goonerDroid There is a LICENSE file in the root, we do not need to add it to each file. Please remove this. Makes it very cluttered. Let Spotless only format the code for readability for now.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (darshanpania) [reply app/src/main/AndroidManifest.xml:2]: @goonerDroid See my comment on the other thread below

File: app/src/main/java/com/darshan/notificity/AppDatabase.kt (hunk 2 above)
Comment (goonerDroid) [reply app/src/main/java/com/darshan/notificity/AppDatabase.kt:1]: Ack

**Comment by darshanpania** [comment]
@goonerDroid Please update your forked code and commit your changes again. The conflicts will be resolved and then I will push out a new version with these changes

**Review by darshanpania — APPROVED** [review]

**Pull request merged**